packages/eddsa-pcd/src/EdDSAPCD.ts

```diff
@@ -6,62 +10,175 @@ import {
+  /**
+   * An EdDSA public key corresponding to the EdDSA private key used 
+   * for signing the message.
+   *
+   * @type {EDdSAPublicKey}
+   */
   publicKey: EDdSAPublicKey;
-  // The message is an array of bigints, each representing a field
-  message: bigint[];
+  
+  /**
+   * An array of contents signed with the private key.
+   * 
+   * @type {Array<bigint>}
+   */
+  message: Array<bigint>;
 }
 
-// The proof is the signature, in the form of a string representation of
-// hex digits.
-// This signature proves that the private key matching publicKey in
-// the claim was used to sign the message.
+/**
+ * This interface defines the arguments to make an EdDSA PCD proof.
+ * The proof is the signature which proves that the private key corresponding to the
+ * public key in the claim, has been successfully used to sign the message.
+ * 
+ * @summary The arguments for an EdDSA PCD proof.
+ * @interface
+ */
 export interface EdDSAPCDProof {
+  /**
+   * The EdDSA signature of the message.
+   * 
+   * @type {string}
+   */
   signature: string;
 }
 
+/**
+ * Create a new EdDSA PCD from an EdDSA PCD claim and proof.
+ * 
+ * @class 
+ * @classdesc This class represents an EdDSA PCD.
+ * @implements {PCD<EdDSAPCDClaim, EdDSAPCDProof>}
+ */
 export class EdDSAPCD implements PCD<EdDSAPCDClaim, EdDSAPCDProof> {
+  /** @public */
   type = EdDSAPCDTypeName;
+
+  /** @public */
+  id: string;
+  /** @public */
   claim: EdDSAPCDClaim;
+  /** @public */
   proof: EdDSAPCDProof;
-  id: string;
 
+  /**
+   * @public
+   * @param {string} id - the unique identifier of the PCD.
+   * @param {EdDSAPCDClaim} claim - the object containing the EdDSA PCD claim.
+   * @param {EdDSAPCDProof} proof - the object containing the EdDSA PCD proof.
+   */
   public constructor(id: string, claim: EdDSAPCDClaim, proof: EdDSAPCDProof) {
     this.id = id;
     this.claim = claim;
     this.proof = proof;
   }
 }
 
+/** 
+ * This variable is a placeholder to check whether the {@link Eddsa} and {@link Poseidon} classes 
+ * have been properly initialized from external library.
+ * 
+ * @see {@link https://github.com/iden3/circomlibjs/blob/main/src/eddsa.js#L9C31-L9C41}
+ */
 let initializedPromise: Promise<void> | undefined;
+/** {@link Eddsa} */
 let eddsa: Eddsa;
+/** {@link Poseidon} */
 let poseidon: Poseidon;
 
+/**
+ * Convert an hexadecimal string to an array Buffer.
+ * 
+ * @function fromHexString
+ * @param {string} hexString - the hexadecimal string to be converted. 
+ * @returns {Buffer} the Buffer representation of the hexadecimal string. 
+ */
+const fromHexString = (hexString: string) => Buffer.from(hexString, "hex");
+
+/**
+ * Convert an array Buffer to an hexadecimal string.
+ * 
+ * @function toHexString
+ * @param {string} bytes - the array Buffer to be converted. 
+ * @returns {string} the hexadecimal string representation of the array Buffer. 
+ */
+const toHexString = (bytes: Uint8Array) => Buffer.from(bytes).toString("hex");
```

Comment:
I think these functions now exist in the `@pcd/util` package. Let's use those instead of these.